Added weather mode

cosmos1978 opened the pull request:
you need to configure your openweathermap api key, city name and country code in the main ino file.
Perhaps the saving can be done better and more inline but the other mqtt component.
Also the json output is very large and because I use HTTPclient.getString it will kill the memory if I make a larger request. 
so what can be improved is the json parser. Perhaps it was better to use jsonserializer. second I need to find a way to parse and store large json objects

vortigont commented:
You can use '/data/2.5/weather?' API call, it gives only current weather, but it is less than 500 bytes bytes long for one city.

cosmos1978 commented:
added weather report on webpage.
added scrolling temperature text on weather effects. ( number is rounded to closest decimal value )
using arduinojson now for all non string values.
